File: 0x06-python-classes/6-square.py
#!/usr/bin/python3
"""Class square create one object with an size define

Raises:
    TypeError: [When size is different of int]
    ValueError: [When size is less that Zero]
    TypeError: [When value is different of int]
    ValueError: [When value is less that Zero]

Returns:
    [Int] -- [attribute private size to public and total area of an square]
"""


class Square:
    """class constructor"""
    def __init__(self, size=0, position=(0, 0)):
        self.size = size
        self.position = position

    """class getter"""
    @property
    def size(self):
        return self.__size

    """class setter"""
    @size.setter
    def size(self, value):
        if type(value) is not int:
            raise TypeError('size must be an integer')
        elif value < 0:
            raise ValueError('size must be >= 0')
        self.__size = value

    """class getter"""
    @property
    def position(self):
        return self.__position

    """class setter"""
    @position.setter
    def position(self, value):
        if len(value) != 2 or type(value) != tuple:
            raise TypeError('position must be a tuple of 2 positive integers')
        elif type(value[1]) != int or type(value[0]) != int:
            raise TypeError('position must be a tuple of 2 positive integers')
        elif value[1] < 0 or value[0] < 0:
            raise ValueError('position must be a tuple of 2 positive integers')
        self.__position = value

    """class for obtain total area of the square"""
    def area(self):
        return self.__size * self.__size

    """method for print object based in his size with space """
    def my_print(self):
        if self.size == 0:
            print()
        else:
            if self.position[1] != 0:
                print('\n' * self.position[1], end="")
            for sqr in range(self.size):
                print(' ' * self.position[0], end="")
                print('#' * self.size)

File: 0x06-javascript-classes/6-square.js
#!/usr/bin/node

module.exports = class Square {
  constructor (size = 0, pst = []) {
    try {
      if (!Number.isInteger(size)) throw 'size must be an integer';
      else if (size < 0) throw 'size must be >= 0';
      else if (typeof (pst) !== 'object') throw 'This datatype not valid';
      else {
        this._size = size;
        this._position = pst;
      }
    } catch (err) {
      console.log(err);
    }
  }

  area () {
    return this._size * this._size;
  }

  get position () {
    return this._position;
  }

  set position (value) {
    try {
      if (typeof (value) !== 'object' || value.length > 2) {
        throw 'position must be a array of 2 positive integers';
      }
    } catch (err) { return console.log(err); }
    this.position = value;
  }

  get size () {
    return this._size;
  }

  set size (value) {
    try {
      if (!Number.isInteger(value)) throw 'size must be an integer';
      else if (value < 0) throw 'size must be >= 0';
      this._size = value;
    } catch (err) {
      console.log(err);
    }
  }

  print () {
    if (this._size === 0) { console.log(' '); } else {
      for (let i = 0; i < this._size; i++) {
        console.log('_'.repeat(this.position[0]) + '#'.repeat(this._size));
      }
    }
  }
};
